zebrad/src/components/mempool/storage/verified_set.rs

```diff
@@ -23,14 +26,26 @@ use zebra_chain::transaction::MEMPOOL_TRANSACTION_COST_THRESHOLD;
 /// This also caches the all the spent outputs from the transactions in the mempool. The spent
 /// outputs include:
 ///
+/// - the dependencies of transactions that spent the outputs of other transactions in the mempool
+/// - the UTXOs of transactions in the mempool
 /// - the transparent outpoints spent by transactions in the mempool
 /// - the Sprout nullifiers revealed by transactions in the mempool
 /// - the Sapling nullifiers revealed by transactions in the mempool
 /// - the Orchard nullifiers revealed by transactions in the mempool
 #[derive(Default)]
 pub struct VerifiedSet {
     /// The set of verified transactions in the mempool.
-    transactions: VecDeque<VerifiedUnminedTx>,
+    transactions: HashMap<transaction::Hash, VerifiedUnminedTx>,
+
+    /// A map of dependencies between transactions in the mempool that
+    /// spend or create outputs of other transactions in the mempool.
+    transaction_dependencies: TransactionDependencies,
+
+    /// The [`transparent::Utxo`]s created by verified transactions in the mempool
+    ///
+    /// Note that these UTXOs may not be unspent.
+    /// Outputs can be spent by later transactions or blocks in the chain.
```

Comment:
Shouldn't we go with outputs instead of UTXOs?

```suggestion
    /// The [`transparent::Output`]s created by verified transactions in the mempool
    ///
    /// Note that these outputs may not be unspent.
    /// They can be spent by later transactions or blocks in the chain.
```

Reply:
Yes we should. Applied in https://github.com/ZcashFoundation/zebra/pull/8857/commits/874372a8bfeabf08ce9aa903e62dfff8b97a566d.